Scripts/RemediationScripts/Remediate-EnableAADForKubernetesService.ps1

```diff
@@ -0,0 +1,564 @@
+        Write-Host "`nDo you want to enable AAD for all Kubernetes Services? " -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if($userInput -ne "Y")
+        {
+            Write-Host "AAD will not be enabled for any Kubernetes Services. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "`n[Step 5 of 5] Enabling AAD for Kubernetes Services..."
+
+        # To hold results from the remediation.
+        $kubernetesClusterRemediated = @()
+        $kubernetesClusterSkipped = @()
+
+        $setAADProfile = [SetAADProfile]::new()
+
+        $kubernetesServicesWithoutAADEnabled | ForEach-Object {
+            $kubernetesServiceResource = $_
+            $resourceId = $_.ResourceId
+            $resourceGroupName = $_.ResourceGroupName
+            $resourceName = $_.ResourceName
+            $location = $_.Location
+            $isRBACEnabled = $_.IsRBACEnabled
+            $aadClientIds = @()
+
+            Write-Host $([Constants]::SingleDashLine)
+            Write-Host "`nEnabling AAD for Kubernetes cluster [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+
+            # Check whether RBAC is enabled on cluster or not. Without RBAC, AAD cannot be enabled.
+            if ($isRBACEnabled)
+            {
+                if ($addAADGroup)
+                {
+                    Write-Host "Please provide Azure AD Group client id: "
+                    $aadClientIds = Read-Host
+                }
+                else
+                {
+                    Write-Host "AAD group will not be added to Kubernetes cluster." -ForegroundColor $([Constants]::MessageType.Info)
+                }
+
+                # Setting AAD profile config to Kubernetes cluster.
+                $res = $setAADProfile.SetAADProfileForCluster($subscriptionId, $resourceName, $resourceGroupName, $location, $aadClientIds)
+
+                if ($res.properties.aadProfile.managed)
+                {
+                    $kubernetesServiceResource.IsAADEnabled = $true
+                    $kubernetesClusterRemediated += $kubernetesServiceResource
+                    Write-Host "Successfully enabled AAD for Kubernetes cluster [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Update)
+                }
+                else
+                {
+                    $kubernetesClusterSkipped += $kubernetesServiceResource
+                    Write-Host "Error occurred enabling AAD for Kubernetes cluster [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Error)
+                }
+            }
+            else
+            {
+                $kubernetesClusterSkipped += $kubernetesServiceResource
```

Comment:
Let's create one more collection of AKSCluster which has RBAC disabled and print them separately in a table instead of one line for each cluster.

Reply:
We have separate field to print RBAC status for each of the AKS cluster.